### ISSUE TITLE
Aws Amplify: Partial updates(mutations)

### ISSUE THREAD

**Issue by ajitgoel** [aws-amplify/amplify-cli#4917]
I have 2 questions that I need input on, from the community:

(1) I have made changes to my application's schema.graphql file. How do i ensure that the corresponding queries.js, mutations.js, subscriptions.js files are updated? Previously these used to get updated, when I ran the `amplify push` command but now they no longer do.

(2) How would I do a partial mutation using aws amplify? eg: if a mutation has fullName, city, how would i update fullName without passing city from the frontend application? Here's how the mutation looks like:

**mutations.js:**

```
const updateUserProfile =
  mutation UpdateUserProfile(
    $input: UpdateUserProfileInput!
    $condition: ModelUserProfileConditionInput
  ) {
    updateUserProfile(input: $input, condition: $condition) {
      id
      fullName
      city
      createdAt
      updatedAt
      owner
    }
  }
;
```
**userprofile.vue**

```
import { Auth } from 'aws-amplify';
import { createUserProfile, updateUserProfile} from '@/graphql/mutations';    
const userProfileInput={
              id:userId, 
              fullName:'Ajit Goel',  
            };
await API.graphql(graphqlOperation(updateUserProfile, {input: userProfileInput}));
```

**schema.graphql:**

```
 type UserProfile @model 
    @key(fields:["id"])
    @auth(rules: [{allow: owner}])
    {
        id: String!
      fullName: String
      city:String
  }
```
**Error in console when update mutation is run:**

![image](https://user-images.githubusercontent.com/6394444/88255900-49d79700-cc7f-11ea-8a2e-1437364dd200.png)

**Comment by dabit3** [comment]
1. Make sure you run `amplify push` from the root of your application. If you run it from within any other folder it will not work.

2. You should not be getting that error if the `city` field is optional. Can you go into the schema located at `amplify/backend/api/you_api_name/build/schema.graphql` and see if the schema compiled to have the `city` be a nullable field (meaning that there is no !)

**Comment by ajitgoel** [comment]
a. i am running "amplify push" from the root of the application but the schema does not compile to have the city. 
b. The "city" field is optional ie there is no !. The schema is not getting compiled to have the city.

**Comment by nikhname** [comment]
@ajitgoel 
1. 
a. Which version of the CLI are you using for the push? 
b. Are you prompted with these two questions after running ```amplify push``` -
```
? Do you want to update code for your updated GraphQL API Yes
? Do you want to generate GraphQL statements (queries, mutations and subscription) based on your schema type
s? This will overwrite your current graphql queries, mutations and subscriptions Yes
```

**Comment by ajitgoel** [comment]
i am using amplify version 4.24.3. 
check the video here: https://www.loom.com/share/e0edae54972c40dab046bfef6e0afd0f
at 2.33 u can see that it does not detect that there are changes to graphql api(3rd from bottom), at 8.20 u can see that there are graphql changes that I made to test and that need to be checked into source control. 

The source code is checked into https://github.com/ajitgoel/VideoAudioTranscription3. It does throw runtime errors(that i need to fix).

**Comment by dabit3** [comment]
Hey, after watching the video I think the changes you made are in the #currentcloudbackend folder. Try making them in the #backend folder.

**Comment by nikhname** [comment]
@ajitgoel yes, you would need to update the schema in your ```/backend/api/API_NAME/``` folder. This could be the cause for the error in (2.) as well, since the ```city``` field would not be compiled in your API when updating #currentcloudbackend

**Comment by ajitgoel** [comment]
so there is something deeper going on here and it is seems to be related with `amplify add env `command. 

Initially(when @dabit3 , @nikhil-dabhade messaged me yesterday), the changes in amplify/backend/api/you_api_name/build/schema.graphql were present in the #currentcloudbackend folder but the application did not work even after doing a `amplify push`(it never asked me to generate and push the amplify graphql changes to aws).
To work around this, i tried to delete the env(deleting env through aws amplify threw errors), create a new envt, delete the existing graphql api, push the changes to aws, create a new graphql api, push the changes to aws (the system asked me to generate and push the amplify graphql changes to aws). the application still does not work. Here's the output for the same: https://docs.google.com/document/d/1L5E80kwmkUBao1JQ067aPKsvHecXQ7Jcduhm_X2XvrA/edit?usp=sharing

here's the console output from my application:

![image](https://user-images.githubusercontent.com/6394444/88713370-d13e6380-d0e0-11ea-81d3-b8ed956d19b1.png)

here's the code that is failing:

import { getUserProfile} from '@/graphql/queries';
async created() 
    {
        const currentUserInfoResult=await this.currentUserInfo();
        const userId=currentUserInfoResult.id;        
        const result = await API.graphql(graphqlOperation(getUserProfile, {id: userId}));        
        console.log(`result: ${JSON.stringify(result)}`);<= it never shows this result. 
    }, 
    
I would very much like AWS amplify to work but I am unsure of what else I can try. Please advise.

**Comment by ajitgoel** [comment]
I just could not get aws amplify to work, specifically graphql api. Some of the things that I tried are 
a. run amplify delete command to delete all resources from aws. 
b. physically deleting the amplify folder from the application in the local pc. 
c. re-initializing the application all the way from aws amplify init to amplify publish. 

I don't remember facing these many issues when I used firebase. Three months of work wasted.
